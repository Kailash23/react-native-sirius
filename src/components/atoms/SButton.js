import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { Button, Text } from 'react-native-paper';

function SButton(props) {
  const { label, onPressAction, buttonContentStyle, labelStyle } = props;
  const handleOnPress = () => {
    if (onPressAction) {
      onPressAction();
    } else {
      return null;
    }
  };
  return (
    <View
      style={[
        {
          flex: 1,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center'
        }
      ]}>
      <TouchableOpacity
        style={[
          buttonContentStyle,
          {
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center'
          }
        ]}
        onPress={() => handleOnPress()}>
        <Text style={labelStyle}>{label}</Text>
      </TouchableOpacity>
    </View>
  );
}

export default SButton;
