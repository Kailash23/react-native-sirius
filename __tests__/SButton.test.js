import SButton from '@atoms/SButton';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { TouchableOpacity } from 'react-native';

configure({ adapter: new Adapter() });

describe('Testing <SButton/>', () => {
  test('SButton should render properly', () => {
    const wrapper = shallow(
      <SButton
        label={'Test button'}
        labelStyle={{
          color: 'red'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('SButton onPress functionlaity should work as intended.', () => {
    const onPressEvent = jest.fn();
    onPressEvent.mockReturnValue('Button on press invoked');
    const wrapper = shallow(
      <SButton
        onPressAction={onPressEvent}
        label={'Test button'}
        labelStyle={{
          color: 'red'
        }}
      />
    );
    expect(onPressEvent).toHaveBeenCalledTimes(0);
    wrapper
      .find(TouchableOpacity)
      .first()
      .props()
      .onPress();
    expect(onPressEvent).toHaveBeenCalledTimes(1);
  });
});
