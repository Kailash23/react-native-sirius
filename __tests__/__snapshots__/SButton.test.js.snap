// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SButton/> SButton should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <ForwardRef
    onPress={[Function]}
    style={
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <withTheme(Text)
      style={
        Object {
          "color": "red",
        }
      }
    >
      Test button
    </withTheme(Text)>
  </ForwardRef>
</View>
`;
